Add mention of tnoremap to map.txt

ssonicblue opened the pull request:
Reference issue #5010 .

Commit:
Add mention of tnoremap to map.txt

fmoralesc commented on runtime/doc/map.txt:
```diff
@@ -71,6 +71,7 @@ modes.
 :ino[remap] {lhs} {rhs}		|mapmode-i|	*:ino* *:inoremap*
 :ln[oremap] {lhs} {rhs}		|mapmode-l|	*:ln*  *:lnoremap*
 :cno[remap] {lhs} {rhs}		|mapmode-c|	*:cno* *:cnoremap*
+:tno[remap] {lhs} {rhs}   |mapmode-t| *:tno* *:tnoremap*
```
There is no *mapmode-t*, so it needs to be added (see lines 291). It should be mentioned also at the end of the list in line 318.

fmoralesc replied:
BTW, there seems to be an alignment issue. Previous entries have an space and a tab, instead of just spaces.

Commit:
Add mention of mapmode-t and terminal mapping

Add mention of mapmode-t to the list of map modes, and add the relevant commands to the map-modes command list.

ssonicblue commented:
How is that? Apologies for my sloppiness, this is my first change to the neovim docs.

ssonicblue commented on runtime/doc/map.txt:
```diff
@@ -289,16 +289,17 @@ as a special key.
+			*mapmode-nvo* *mapmode-n* *mapmode-v* *mapmode-o* *mapmode-t*
 
-There are six sets of mappings
+There are seven sets of mappings
```
How is this? Sorry for my sloppiness, this is my first change to neovim's docs.

fmoralesc replied:
:+1:

justinmk commented on runtime/doc/map.txt:
```diff
@@ -525,9 +528,9 @@ scenario: >
+byte.  If you type the character á (0xe1 <M-a>) in UTF-8 encoding this is the
 two bytes 0xc3 0xa1.  You don't want the 0xc3 byte to be mapped then or
-otherwise it would be impossible to type the � character.
+otherwise it would be impossible to type the á character.
```
encoding problem.
